Sources/SwiftSyntax/SyntaxCollection.swift

```diff
@@ -33,6 +33,11 @@ extension SyntaxCollection {
     self.init(Syntax(data))!
   }
 
+  /// Create an empty ``SyntaxCollection`` with no children.
+  public init() {
+    self.init([])
```

Comment:
I guess this is the other downside of using `RangeReplaceableCollection`. This constructor should really never be used and makes it super easy to drop the parent accidentally.